proposals/hasCrossSiteAncestor.md

```diff
@@ -0,0 +1,91 @@
+# Proposal: Update Extensions to interact with Partitioned Cookies containing Cross-Site Ancestor Chain Bit values
+
+**Summary**
+
+Allow extensions to utilize a hasCrossSiteAncestor boolean value when interacting with partitioned cookies that include a cross-site ancestor chain bit in their partiton key.
+
+**Document Metadata**
+
+**Author:**  aselya 
+
+**Sponsoring Browser:** Chrome
+
+**Contributors:** DCtheTall
+
+**Created:** 2024-04-01
+
+**Related Issues:** [How do Partitioned cookies interact with browser extensions?](https://github.com/privacycg/CHIPS/issues/6)
+
+## Motivation
+[Chrome](https://github.com/privacycg/CHIPS/issues/40) is adding a cross-site ancestor bit value to partitioned cookies. 
+
+### Objective
+To interact with partitioned cookies containing a cross-site ancestor chain bit correctly, extensions will need to have the ability to to specify a value (hasCrossSiteAncestor) that cooresponds to the value of the cross-site ancestor chain bit in partitioned cookies.
```

Comment:
If I'm understanding correctly, this bit makes it no longer possible to enumerate all cookies for a given origin (since there isn't a way to discover all of the possible first-party sets that could exist).

Am I understanding correctly and if so, is that something we are concerned about? It would be good to address that and either propose a new API method to resolve this or otherwise add a brief note on "non-goals" / issues we aren't attempting to solve. I don't have strong feelings either way.

Reply:
The ability to enumerate all cookies for a given origin should still be possible using `cookies.GetAll()` 

If a developer wants to access both partitioned and unpartitioned cookies:
A developer can pass a partitionKey without a topLevelSite and hasCrossSiteAncestor values to the `cookies.GetAll()` function and it will retrieve all cookies from a single cookie store that match the other information contained in the details object.

If a developer wants to access partitioned cookies regardless of the hasAncestorChainBit value:
A developer can pass a partitionKey with a topLevelSite but no hasCrossSiteAncestor value to the `cookies.GetAll()` function. That call will return all partitioned cookies that have the same topLevelSite and match the other information contained in the details object.

@oliverdunk, let me know if this clarification doesn't address your concern.

Reply:
Thanks for taking the time to explain!

> If a developer wants to access partitioned cookies regardless of the hasAncestorChainBit value: A developer can pass a partitionKey with a topLevelSite but no hasCrossSiteAncestor value to the cookies.GetAll() function.

Ah, this makes sense. I missed that the behavior for cookies.get and cookies.getAll were different and getAll does not have the same logic to try and deduce a value for `hasCrossSiteAncestor`.

Reply:
No problem at all, glad glad to clarify.